db/migrate/20170927135007_migrate_dialog_field_associations_to_use_new_relationship.rb

```diff
@@ -0,0 +1,62 @@
+class MigrateDialogFieldAssociationsToUseNewRelationship < ActiveRecord::Migration[5.0]
+  class DialogFieldAssociation < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogField < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class Dialog < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogTab < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogGroup < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+
+  def up
+    dialog_fields_with_associations = DialogField.all.select { |df| df.auto_refresh || df.trigger_auto_refresh }
+    absolute_position = {}
+    unless dialog_fields_with_associations.blank?
+      dialog_fields_with_associations.each do |f|
+        index =
+          f.position +
+          DialogGroup.find(f.dialog_group_id).position * 100 +
+          DialogTab.find(DialogGroup.find(f.dialog_group_id).dialog_tab_id).position * 1_000 +
+          DialogTab.find(DialogGroup.find(DialogField.find(f.id).dialog_group_id).dialog_tab_id).dialog_id * 1_000_000
```

Comment:
Why is this one needed?

Reply:
Because we can have two tabs with the same position and I need to check to make sure the fields are on the same dialog anyway so why not do it there?